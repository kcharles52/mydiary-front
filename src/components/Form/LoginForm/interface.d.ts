export interface LoginProps {
  onChange: (fieldName: string, value: string) => void;
  onSubmit: (form:string) => void;
  [key:string]:any;
}
