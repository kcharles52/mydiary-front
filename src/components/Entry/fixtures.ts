export const props = {
  entry_id: 1,
  title: "Title 1",
  diaryBody:
    "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
  date: "12/12/2018",
  delete:jest.fn()

};
