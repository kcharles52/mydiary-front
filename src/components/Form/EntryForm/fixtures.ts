export const EntryFormProps = {
  Edit: false,
  title: "string",
  diaryTitle: "string",
  date: "string",
  diaryBody: "string",
  onSubmit: jest.fn(),
  onCancel: jest.fn(),
  onChange: jest.fn()
};

  