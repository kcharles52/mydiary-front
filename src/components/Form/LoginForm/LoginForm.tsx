//react libraries
import * as React from "react";

//components
import { Input, Button } from "../common";

//interfaces
import { LoginProps } from "./interface";
export const LoginForm: React.SFC<LoginProps> = props => {
  return (
    <form action="">
      <h1>Welcome Back</h1>
      <Input
        name="email"
        type="text"
        label="Email Address *"
        value={props.email||""}
        onChange={props.onChange}
      />
      <Input
        name="password"
        label="Password *"
        type="password"
        value={props.password||""}
        onChange={props.onChange}
      />
      <Button
        label="Login"
        className="button button-block"
        onClick={()=>{props.onSubmit("login")}}
      />
    </form>
  );
};
