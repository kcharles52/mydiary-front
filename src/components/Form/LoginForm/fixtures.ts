export const LoginFormProps = {
    onChange: jest.fn(),
    onSubmit: jest.fn(),
  }
  