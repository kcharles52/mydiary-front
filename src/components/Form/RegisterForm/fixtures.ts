export const RegisterFormProps = {
    onChange: jest.fn(),
    onSubmit: jest.fn(),
  }
  