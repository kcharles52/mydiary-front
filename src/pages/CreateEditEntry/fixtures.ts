export const props = {
  Edit: false,
  title: "Form title",
  diaryTitle: "Test Diary Title",
  date: "20-2-19",
  diaryBody: "Test Diary body",
  onSubmit: jest.fn(),
  onChange: jest.fn(),
  addEntryAction: jest.fn(),
  editEntryAction: jest.fn(),
  getEntryAction: jest.fn(() =>
    Promise.resolve({
      entry: {
        entry_id: 1,
        diaryTitle: "Title 1",
        diaryEntryBody:
          "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
        date: "12/12/2018",
        delete: jest.fn()
      }
    })
  ),
  match: { params: { entry_id: "1" } },
  editEntry: {
    entry: {
      entry_id: 1,
      diaryTitle: "Title 1",
      diaryEntryBody:
        "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
      date: "12/12/2018",
      delete: jest.fn()
    }
  },
  newEntry: {},
  history: { push: () => {} },
  location: { pathname: "/addEntry" }
};

export const entry = {
  entry_id: 1,
  diaryTitle: "Title 1",
  diaryEntryBody:
    "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
  date: "12/12/2018",
  delete: jest.fn()
};
