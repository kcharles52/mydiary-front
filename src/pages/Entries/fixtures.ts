export const props = {
  getEntriesAction: jest.fn(),
  deleteEntryAction: jest.fn(),
  entries: [
    {
      entry_id: 1,
      diaryTitle: "Title 1",
      diaryEntryBody:
        "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
      date: "12/12/2018",
      delete: jest.fn()
    },
    {
      entry_id: 2,
      diaryTitle: "Title 2",
      diaryEntryBody:
        "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
      date: "12/12/2018",
      delete: jest.fn()
    }
  ],
  deleted: [],
  handlesDelete: jest.fn(),
  history: {push:jest.fn()},
  location: { pathname: "/home" },
  match: { url: "/home" }
};

export const fetchedEntries =[
  {
    entry_id: 1,
    diaryTitle: "Title 1",
    diaryEntryBody:
      "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
    date: "12/12/2018",
    delete: jest.fn()
  },
  {
    entry_id: 2,
    diaryTitle: "Title 2",
    diaryEntryBody:
      "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
    date: "12/12/2018",
    delete: jest.fn()
  }
]