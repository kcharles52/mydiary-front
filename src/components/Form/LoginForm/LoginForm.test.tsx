// react libraries
import * as React from "react";

// third party libraries
import { mount, shallow } from "enzyme";

// components
import { LoginForm } from "./LoginForm";

//fixtures
import { LoginFormProps as props } from "./fixtures";

describe("Login form component", () => {
  let wrapper, shallowWrapper;

  beforeEach(() => {
    wrapper = mount(<LoginForm {...props} />);
    shallowWrapper = shallow(<LoginForm {...props} />);
  });
  afterEach(() => {
    wrapper = shallowWrapper = null;
  });

  it("should render login form component", () => {
    expect(shallowWrapper).toMatchSnapshot();
    expect(shallowWrapper.find("Input").length).toBe(2);
  });

  it("should login user", () => {
    wrapper.find("Button").simulate("click");
    expect(wrapper.prop("onSubmit")).toBeCalled()
  });
});
