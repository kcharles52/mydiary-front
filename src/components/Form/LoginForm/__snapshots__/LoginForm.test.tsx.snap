// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login form component should render login form component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginForm
    onChange={[MockFunction]}
    onSubmit={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "action": "",
      "children": Array [
        <h1>
          Welcome Back
        </h1>,
        <Input
          label="Email Address *"
          name="email"
          onChange={[MockFunction]}
          type="text"
          value=""
        />,
        <Input
          label="Password *"
          name="password"
          onChange={[MockFunction]}
          type="password"
          value=""
        />,
        <Button
          className="button button-block"
          label="Login"
          onClick={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Welcome Back",
        },
        "ref": null,
        "rendered": "Welcome Back",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "label": "Email Address *",
          "name": "email",
          "onChange": [MockFunction],
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "label": "Password *",
          "name": "password",
          "onChange": [MockFunction],
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "button button-block",
          "label": "Login",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "action": "",
        "children": Array [
          <h1>
            Welcome Back
          </h1>,
          <Input
            label="Email Address *"
            name="email"
            onChange={[MockFunction]}
            type="text"
            value=""
          />,
          <Input
            label="Password *"
            name="password"
            onChange={[MockFunction]}
            type="password"
            value=""
          />,
          <Button
            className="button button-block"
            label="Login"
            onClick={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome Back",
          },
          "ref": null,
          "rendered": "Welcome Back",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "label": "Email Address *",
            "name": "email",
            "onChange": [MockFunction],
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "label": "Password *",
            "name": "password",
            "onChange": [MockFunction],
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "button button-block",
            "label": "Login",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
